### DISCUSSION TITLE
Should react-aria hooks expose data attributes in props?

### DISCUSSION THREAD

**Discussion by Gomah** [adobe/react-spectrum#5632] [Q&A] [answered]
The Button component from `react-aria-components` exposes the data attributes by default (focus, hovered, pressed etc), which is great 👍🏻 

However, when customising the component (as recommended here: https://react-spectrum.adobe.com/react-aria/Button.html#hooks ), I'm not getting the data attributes when using `useButton` from `react-aria`, is it intended?

What would be the right approach if I wanted to keep these data attributes & apply them to a custom component? (Like you would do with polymorphism or a `asChild` approach like [Radix UI Slots](https://www.radix-ui.com/primitives/docs/utilities/slot))

Repro: 
https://codesandbox.io/p/sandbox/aria-button-props-crkl4z

**Comment by snowystinger** [comment, marked as answer]
Hey, thanks for the discussion. We have soft decided not to include the data-attributes because it would be more opinionated and the hooks are meant to be general.

We do recognize that it might make the "dropping down" to the hooks, from RAC specifically, a bit easier, but figured we'd hold off in the initial release to see how common it is in practice.

The right approach right now would be to add the appropriate hooks and create the data-attributes yourself. You can look at the source or docs for a given RAC component in order to see which ones should be applied as well as how we apply it ourselves.

This approach also allows people to more easily customize the name of the `data-*` attributes in case they conflict with something already being used in your library.

**Comment by Gomah** [comment]
Thanks @snowystinger – makes sense!

Maybe the "Advanced customization" section in the docs should include [extra details](https://github.com/adobe/react-spectrum/blob/main/packages/react-aria-components/src/Button.tsx#L82-L83) to mimic the RAC behaviour / props ?
